src/bin/zmcontactbackup

```diff
@@ -0,0 +1,565 @@
+    if ( !$comments_good ) {
+        print STDERR "Rebuilding contact backup cron\n\n";
+
+        # Find ZIMBRAEND, and add contact backup comments
+        # before it. If not found add comments at the end
+        my $zimbraEndLoc = 0;
+        for ( my $i = 0 ; $i <= $#cron ; $i++ ) {
+            $_ = $cron[$i];
+            if (
+m/ZIMBRAEND -- DO NOT EDIT ANYTHING BETWEEN THIS LINE AND ZIMBRASTART/
+              )
+            {
+                $zimbraEndLoc = $i;
+                last;
+            }
+        }
+        if ($zimbraEndLoc) {
+
+            # ZIMBRAEND found, insert contact backup comments before ZIMBRAEND
+            splice(
+                    @cron,
+                    $zimbraEndLoc - 1,
+                    0,
+                    (
+                       "#\n# CONTACT BACKUP BEGIN\n",
+                       "# CONTACT BACKUP END\n#\n"
+                    )
+            );
+            saveCron();
+            loadCron();
+        } else {
+
+            # ZIMBRAEND not found
+            # One or both contact backup comments not found.
+            # Clean up the array, and add them to the end
+            if ( $cronstart == -1 && $cronstop == -1 ) {
+
+                # No comments at all
+                push @cron, "#\n# CONTACT BACKUP BEGIN\n";
+                $cronstart = $#cron;
+                $cronstop  = $#cron;
+                push @cron, "# CONTACT BACKUP END\n#\n";
+                saveCron();
+                loadCron();
+            } else {
+
+           # It's not possible to find an end and no start.
+           # No end comment - add the end comment right after the start comment.
+                splice( @cron, $cronstart + 1, 0, "# CONTACT BACKUP END\n" );
+            }
+        }
+    }
+}
+
+sub displayCurrentSchedule {
+    print "Current Schedule:\n\n";
+    for ( my $i = $cronstart + 1 ; $i <= $cronstop ; $i++ ) {
+        $_ = $cron[$i];
+        my @fields = split;
+        my $cnt    = @fields;
+        if ( $cnt < 9 ) {
+            print STDERR
+              "Invalid schedule found. Rebuild contact backup cron.\n\n";
+        }
+        print "\t$fields[0] $fields[1] $fields[2] $fields[3] $fields[4]";
+        my $tmp = $fields[8];
+        $tmp = substr( $tmp, ( index( $tmp, "=" ) + 1 ) );
+        print " $tmp";
+        if ( $cnt > 9 ) {
+            for my $j ( 9 ... ( $cnt - 1 ) ) {
+                $tmp = $fields[$j];
+                if ( index( $tmp, "server=" ) != -1 ) {
+                    $tmp = substr( $tmp, ( index( $tmp, "=" ) + 1 ) );
+                    print " $tmp";
+                }
+            }
+        }
+        print "\n\n";
+    }
+    exit(0);
+}
+
+sub prepareSoap {
+    $soap = "/opt/zimbra/bin/zmsoap -z ContactBackupRequest \@op=";
+    if ($stop) {
+        $soap .= "stop";
+    } else {
+        $soap .= "start";
+    }
+    my $pref = " servers ";
+    foreach my $srv (@servers) {
+        $soap .= $pref . "server=" . $srv . " \@by=name";
```

Comment:
nit: we don't sanitize our server list... if we were paranoid, we would as we're going to use these as arguments to a shell command